services/keybase/keybase-btc.service.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const KeybaseProfile = require('./keybase-profile')
+const Joi = require('joi')
+
+const bitcoinAddressFoundSchema = Joi.object({
+  them: Joi.array()
+    .items(
+      Joi.object({
+        cryptocurrency_addresses: {
+          bitcoin: Joi.array()
+            .items(
+              Joi.object({
+                address: Joi.string().required(),
+              }).required()
+            )
+            .required(),
+        },
+      }).required()
+    )
+    .min(0)
+    .max(1),
+}).required()
+
+const profileNotFoundSchema = Joi.object({
+  them: Joi.array().empty(),
+}).required()
+
+const bitcoinAddressFoundOrNotFound = Joi.alternatives(
+  bitcoinAddressFoundSchema,
+  profileNotFoundSchema
+)
+
+module.exports = class KeybaseBTC extends KeybaseProfile {
+  static get apiVersion() {
+    return '1.0'
+  }
+
+  static get category() {
+    return 'social'
+  }
+
+  static get route() {
+    return {
+      base: 'keybase/btc',
+      pattern: ':username',
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'btc' }
+  }
+
+  async handle({ username }) {
+    const options = {
+      method: 'GET',
+      form: {
+        usernames: username,
+        fields: 'cryptocurrency_addresses',
+      },
+    }
+
+    const data = await this.fetch({
+      schema: bitcoinAddressFoundOrNotFound,
+      options,
+    })
+
+    try {
+      const address = data.them[0].cryptocurrency_addresses.bitcoin[0].address
+      return this.constructor.render({ address })
+    } catch (err) {
+      return {
```

Comment:
How common a scenario will the `bitcoin` array be empty? If it's something that could occur relatively frequently then it's better to handle it just like you have here (allows for providing the user a better error message), but if there really should be an entry then this could be enforced in the schema validation, and we could remove the `try/catch` block

Reply:
It's common actually. Lots of Keybase profiles don't have any bitcoin addresses.

Reply:
👍